common/djangoapps/util/tests/test_markup.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+"""
+Tests for util.markup
+"""
+
+import unittest
+
+import ddt
+
+from edxmako.template import Template
+from util.markup import escape, HTML, ugettext as _, ungettext
+
+
+@ddt.ddt
+class FormatHtmlTest(unittest.TestCase):
+    """Test that we can format plain strings and HTML into them properly."""
+
+    @ddt.data(
+        (u"hello", u"hello"),
+        (u"<hello>", u"&lt;hello&gt;"),
+        (u"It's cool", u"It&#39;s cool"),
+        (u'"cool," she said.', u'&#34;cool,&#34; she said.'),
+        (u"Stop & Shop", u"Stop &amp; Shop"),
+        (u"<a>нтмℓ-єѕ¢αρє∂</a>", u"&lt;a&gt;нтмℓ-єѕ¢αρє∂&lt;/a&gt;"),
+    )
+    def test_simple(self, (before, after)):
+        self.assertEqual(unicode(_(before)), after)  # pylint: disable=translation-of-non-string
+        self.assertEqual(unicode(escape(before)), after)
+
+    def test_formatting(self):
+        # The whole point of this function is to make sure this works:
+        out = _(u"Point & click {start}here{end}!").format(
```

Comment:
Can't we just exclude tests/ directory from the scraping? I'm nearly
certain we do this already, just good to double check

On Tue, Dec 15, 2015 at 10:41 AM, Ned Batchelder <notifications@github.com>
wrote:

> In common/djangoapps/util/tests/test_markup.py
> <https://github.com/edx/edx-platform/pull/10983#discussion_r47652400>:
>
> > +
> > +    @ddt.data(
> > +        (u"hello", u"hello"),
> > +        (u"<hello>", u"&lt;hello&gt;"),
> > +        (u"It's cool", u"It&#39;s cool"),
> > +        (u'"cool," she said.', u'&#34;cool,&#34; she said.'),
> > +        (u"Stop & Shop", u"Stop &amp; Shop"),
> > +        (u"<a>нтмℓ-єѕ¢αρє∂</a>", u"&lt;a&gt;нтмℓ-єѕ¢αρє∂&lt;/a&gt;"),
> > +    )
> > +    def test_simple(self, (before, after)):
> > +        self.assertEqual(unicode(_(before)), after)  # pylint: disable=translation-of-non-string
> > +        self.assertEqual(unicode(escape(before)), after)
> > +
> > +    def test_formatting(self):
> > +        # The whole point of this function is to make sure this works:
> > +        out = _(u"Point & click {start}here{end}!").format(
>
> If it is being scraped, we can try something subversive to prevent it,
> like (_)(u"...")
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/10983/files#r47652400>.
>